File: LAB QUIZZES/CSE221_Quiz01/_A1F/_A1F_Testing.java
import java.io.*;
import java.util.*;

// Compile and run: javac _A1F_Testing.java && java _A1F_Testing

class _A1F_Solution {
    public static class Pair<K, V> {
        public K key;
        public V value;

        public Pair(K key, V value) {
            this.key = key;
            this.value = value;
        }

        public String toString() {
            return "(" + key + ", " + value + ")";
        }
    }

    public static Pair<Integer, ArrayList<Pair<Integer, Integer>>> solve(ArrayList<Integer> sIDs,
            ArrayList<Integer> sMarks) {
        int N = sIDs.size(), cnt = 0;
        ArrayList<Pair<Integer, Integer>> arr = new ArrayList<>();
        // Place your code here
        return new Pair<>(cnt, arr);
    }
}

public class _A1F_Testing {
    Random rng = new Random();
    public static int N = 0;
    public static ArrayList<Integer> sIDs = new ArrayList<>(), sMarks = new ArrayList<>();
    public static _A1F_Solution.Pair<Integer, ArrayList<_A1F_Solution.Pair<Integer, Integer>>> res;

    public static void main(String args[]) throws Exception {
        // // stdin
        // // 7
        // // 7 4 9 3 2 5 1
        // // 40 50 50 20 10 10 10
        BufferedReader fin = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(fin.readLine().replace("\\s", ""));
        for (String temp : fin.readLine().split(" "))
        sIDs.add(Integer.parseInt(temp));
        for (String temp : fin.readLine().split(" "))
        sMarks.add(Integer.parseInt(temp));


        long start = System.nanoTime();
        res = _A1F_Solution.solve(new ArrayList<>(sIDs), new ArrayList<>(sMarks));
        long finish = System.nanoTime();
        long elapsed = finish - start;


        // // stdout
        // System.out.println(sIDs.toString());

        System.out.printf("Execution time for the targeted one test is %.9fs\n", elapsed * 1e-9);
        System.exit(0);

        // // file input
        // BufferedReader fin = new BufferedReader(new
        // FileReader("TimeLimitExceeded.txt"));
        // BufferedReader fin = new BufferedReader(new FileReader("CustomTest.txt"));
        // BufferedReader fin = new BufferedReader(new FileReader("WrongAnswer.txt"));
        // for (int i = 1; i <= 3; ++i)
        //     fin.readLine();
        // fin.skip(8);
        // for (String temp : fin.readLine().replace("]", "").split(", "))
        //     sIDs.add(Integer.parseInt(temp));
        // fin.skip(10);
        // for (String temp : fin.readLine().replace("]", "").split(", "))
        //     sIDs.add(Integer.parseInt(temp));
        // fin.close();

        // // hardcoded input
        // sIDs = new ArrayList<>(Arrays.asList(7, 4, 9, 3, 2, 5, 1));
        // sMarks = new ArrayList<>(Arrays.asList(40, 50, 50, 20, 10, 10, 10));

        
        // // file output
        // BufferedWriter fout = new BufferedWriter(new FileWriter("OutputOfYourCode.txt"));
        // fout.write(sIDs.toString());
        // fout.close();
    }
}

File: LAB QUIZZES/CSE221_Quiz01/_A1F/_A1F_Testing.py
import os, sys, time, random, threading, subprocess

# Compile and run: pypy _A1F_Testing.py
def solve(sIDs, sMarks):
    N = len(sIDs)
    cnt = 0
    arr = []
    print( sIDs)
    for i in range(N):
        max_idx = i
        max = sMarks[i]
        flag = False
        print("____________")
        print("LOOP", i)
        for j in range(i+1, N):
            if sMarks[j] > max:
                max_idx = j
                max = sMarks[j]
                flag = True
            elif sMarks[j] == max:
                if sIDs[j]<sIDs[max_idx]:
                    max_idx = j
                    flag = True
        if flag == True:
            print(sMarks[i], sMarks[max_idx])
            sMarks[i], sMarks[max_idx] = sMarks[max_idx], sMarks[i]
            sIDs[i], sIDs[max_idx] = sIDs[max_idx], sIDs[i]
            print(sMarks)
            cnt += 1
        print(sMarks)    
    for i, j in zip(sIDs, sMarks):
        id, mark = i, j
        arr.append((id, mark))
    return (cnt, arr)

N = 0
sIDs, sMarks, res = [], [], []

# stdin
# 7
# 7 4 9 3 2 5 1
# 40 50 50 20 10 10 10
N = int(input())
sIDs = list(map(int, input().split()))
sMarks = list(map(int, input().split()))
start = time.time()
res = solve(sIDs.copy(), sMarks.copy())
print(res)
finish = time.time()
elapsed = finish - start

# # stdout
# print(res)

print("Execution time for the targeted one test is " + f"{elapsed:.9f}" + "s", flush=True)
os._exit(0)

# # file input
# with open("TimeLimitExceeded.txt") as fin:
# with open("CustomTest.txt") as fin:
# with open("WrongAnswer.txt") as fin:
#     sIDs = list(map(int, fin.readlines()[3][8:-2].split(", ")))
#     sMarks = list(map(int, fin.readlines()[3][10:-2].split(", ")))

# # hardcoded input
# sIDs = [7, 4, 9, 3, 2, 5, 1]
# sMarks = [40, 50, 50, 20, 10, 10, 10]


# # file output
# with open("OutputOfYourCode.txt", "w") as fout:
#     fout.write(str(res))